<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="7" passed="3" failed="4" skipped="0">
  <reporter-output>
    <line>
      <![CDATA[🚀 Starting Test Suite: Surefire test]]>
    </line>
    <line>
      <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.89
]]>
    </line>
    <line>
      <![CDATA[🟢 Test Started: verifyAccountLockoutAfterMultipleFailedLoginAttempts (🔢 Test Case ID: TC-027)]]>
    </line>
    <line>
      <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
    </line>
    <line>
      <![CDATA[❌ Test FAILED: verifyAccountLockoutAfterMultipleFailedLoginAttempts - Account was not locked after multiple failed login attempts. expected [true] but found [false]]]>
    </line>
    <line>
      <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-027\TC-027_screenshot.png' height='300' width='400'/><br>]]>
    </line>
    <line>
      <![CDATA[🟢 Test Started: verifyChangePasswordFromDashboard (🔢 Test Case ID: TC-118)]]>
    </line>
    <line>
      <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
    </line>
    <line>
      <![CDATA[✅ Test PASSED: verifyChangePasswordFromDashboard (Duration: 35755ms)]]>
    </line>
    <line>
      <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-118\TC-118_screenshot.png' height='300' width='400'/><br>]]>
    </line>
    <line>
      <![CDATA[🟢 Test Started: verifyHTTPSProtocol (🔢 Test Case ID: TC-292)]]>
    </line>
    <line>
      <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
    </line>
    <line>
      <![CDATA[✅ Test PASSED: verifyHTTPSProtocol (Duration: 19358ms)]]>
    </line>
    <line>
      <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-292\TC-292_screenshot.png' height='300' width='400'/><br>]]>
    </line>
    <line>
      <![CDATA[🟢 Test Started: verifyPasswordStrengthRequirements (🔢 Test Case ID: TC-034)]]>
    </line>
    <line>
      <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
    </line>
    <line>
      <![CDATA[❌ Test FAILED: verifyPasswordStrengthRequirements - Error message did not include 'at least 8 characters' requirement. expected [true] but found [false]]]>
    </line>
    <line>
      <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-034\TC-034_screenshot.png' height='300' width='400'/><br>]]>
    </line>
    <line>
      <![CDATA[🟢 Test Started: verifyPaymentGatewaySecurity (🔢 Test Case ID: TC-091)]]>
    </line>
    <line>
      <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
    </line>
    <line>
      <![CDATA[❌ Test FAILED: verifyPaymentGatewaySecurity - Card number is stored in plaintext. It should be encrypted or masked. expected [false] but found [true]]]>
    </line>
    <line>
      <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-091\TC-091_screenshot.png' height='300' width='400'/><br>]]>
    </line>
    <line>
      <![CDATA[🟢 Test Started: verifyProtectionAgainstXSS (🔢 Test Case ID: TC-126)]]>
    </line>
    <line>
      <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
    </line>
    <line>
      <![CDATA[❌ Test FAILED: verifyProtectionAgainstXSS - XSS attack successful: payload is reflected in the Email field. expected [false] but found [true]]]>
    </line>
    <line>
      <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-126\TC-126_screenshot.png' height='300' width='400'/><br>]]>
    </line>
    <line>
      <![CDATA[🟢 Test Started: verifySessionHandling (🔢 Test Case ID: TC-021)]]>
    </line>
    <line>
      <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
    </line>
    <line>
      <![CDATA[✅ Test PASSED: verifySessionHandling (Duration: 17272ms)]]>
    </line>
    <line>
      <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-021\TC-021_screenshot.png' height='300' width='400'/><br>]]>
    </line>
  </reporter-output>
  <suite started-at="2025-03-25T01:15:23 EET" name="Surefire suite" finished-at="2025-03-25T01:20:51 EET" duration-ms="328129">
    <groups>
      <group name="security">
        <method signature="SecurityTest.verifyAccountLockoutAfterMultipleFailedLoginAttempts()[pri:0, instance:tests.ui.SecurityTest@305f7627]" name="verifyAccountLockoutAfterMultipleFailedLoginAttempts" class="tests.ui.SecurityTest"/>
        <method signature="SecurityTest.verifyChangePasswordFromDashboard()[pri:0, instance:tests.ui.SecurityTest@305f7627]" name="verifyChangePasswordFromDashboard" class="tests.ui.SecurityTest"/>
        <method signature="SecurityTest.verifyPasswordStrengthRequirements()[pri:0, instance:tests.ui.SecurityTest@305f7627]" name="verifyPasswordStrengthRequirements" class="tests.ui.SecurityTest"/>
        <method signature="SecurityTest.verifyPaymentGatewaySecurity()[pri:0, instance:tests.ui.SecurityTest@305f7627]" name="verifyPaymentGatewaySecurity" class="tests.ui.SecurityTest"/>
        <method signature="SecurityTest.verifyProtectionAgainstXSS()[pri:0, instance:tests.ui.SecurityTest@305f7627]" name="verifyProtectionAgainstXSS" class="tests.ui.SecurityTest"/>
        <method signature="SecurityTest.verifySessionHandling()[pri:0, instance:tests.ui.SecurityTest@305f7627]" name="verifySessionHandling" class="tests.ui.SecurityTest"/>
      </group> <!-- security -->
      <group name="compliance">
        <method signature="SecurityTest.verifyHTTPSProtocol()[pri:0, instance:tests.ui.SecurityTest@305f7627]" name="verifyHTTPSProtocol" class="tests.ui.SecurityTest"/>
      </group> <!-- compliance -->
      <group name="smoke">
        <method signature="SecurityTest.verifySessionHandling()[pri:0, instance:tests.ui.SecurityTest@305f7627]" name="verifySessionHandling" class="tests.ui.SecurityTest"/>
      </group> <!-- smoke -->
    </groups>
    <test started-at="2025-03-25T01:15:23 EET" name="Surefire test" finished-at="2025-03-25T01:20:51 EET" duration-ms="328129">
      <class name="tests.ui.SecurityTest">
        <test-method is-config="true" signature="setUp(org.testng.ITestResult,org.testng.ITestContext)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:15:23 EET" name="setUp" finished-at="2025-03-25T01:15:42 EET" duration-ms="18455" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name={null} status=CREATED method=SecurityTest.verifyAccountLockoutAfterMultipleFailedLoginAttempts()[pri:0, instance:tests.ui.SecurityTest@305f7627] output={null}]]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[org.testng.TestRunner@6796a873]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="verifyAccountLockoutAfterMultipleFailedLoginAttempts()[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:15:42 EET" name="verifyAccountLockoutAfterMultipleFailedLoginAttempts" description="TC-027" finished-at="2025-03-25T01:17:14 EET" duration-ms="92522" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Account was not locked after multiple failed login attempts. expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Account was not locked after multiple failed login attempts. expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertTrue(Assert.java:57)
at tests.ui.SecurityTest.verifyAccountLockoutAfterMultipleFailedLoginAttempts(SecurityTest.java:102)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:155)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:102)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:91)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:137)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[🟢 Test Started: verifyAccountLockoutAfterMultipleFailedLoginAttempts (🔢 Test Case ID: TC-027)]]>
            </line>
            <line>
              <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
            </line>
            <line>
              <![CDATA[❌ Test FAILED: verifyAccountLockoutAfterMultipleFailedLoginAttempts - Account was not locked after multiple failed login attempts. expected [true] but found [false]]]>
            </line>
            <line>
              <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-027\TC-027_screenshot.png' height='300' width='400'/><br>]]>
            </line>
          </reporter-output>
        </test-method> <!-- verifyAccountLockoutAfterMultipleFailedLoginAttempts -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:17:14 EET" name="tearDown" finished-at="2025-03-25T01:17:15 EET" duration-ms="243" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verifyAccountLockoutAfterMultipleFailedLoginAttempts status=FAILURE method=SecurityTest.verifyAccountLockoutAfterMultipleFailedLoginAttempts()[pri:0, instance:tests.ui.SecurityTest@305f7627] output=🟢 Test Started: verifyAccountLockoutAfterMultipleFailedLoginAttempts (🔢 Test Case ID: TC-027)]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(org.testng.ITestResult,org.testng.ITestContext)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:17:15 EET" name="setUp" finished-at="2025-03-25T01:17:31 EET" duration-ms="16904" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name={null} status=CREATED method=SecurityTest.verifyChangePasswordFromDashboard()[pri:0, instance:tests.ui.SecurityTest@305f7627] output={null}]]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[org.testng.TestRunner@6796a873]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="verifyChangePasswordFromDashboard()[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:17:31 EET" name="verifyChangePasswordFromDashboard" description="TC-118" finished-at="2025-03-25T01:18:07 EET" duration-ms="35755" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[🟢 Test Started: verifyChangePasswordFromDashboard (🔢 Test Case ID: TC-118)]]>
            </line>
            <line>
              <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
            </line>
            <line>
              <![CDATA[✅ Test PASSED: verifyChangePasswordFromDashboard (Duration: 35755ms)]]>
            </line>
            <line>
              <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-118\TC-118_screenshot.png' height='300' width='400'/><br>]]>
            </line>
          </reporter-output>
        </test-method> <!-- verifyChangePasswordFromDashboard -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:18:07 EET" name="tearDown" finished-at="2025-03-25T01:18:08 EET" duration-ms="276" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verifyChangePasswordFromDashboard status=SUCCESS method=SecurityTest.verifyChangePasswordFromDashboard()[pri:0, instance:tests.ui.SecurityTest@305f7627] output=🟢 Test Started: verifyChangePasswordFromDashboard (🔢 Test Case ID: TC-118)]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(org.testng.ITestResult,org.testng.ITestContext)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:18:08 EET" name="setUp" finished-at="2025-03-25T01:18:24 EET" duration-ms="16518" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name={null} status=CREATED method=SecurityTest.verifyHTTPSProtocol()[pri:0, instance:tests.ui.SecurityTest@305f7627] output={null}]]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[org.testng.TestRunner@6796a873]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="verifyHTTPSProtocol()[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:18:24 EET" name="verifyHTTPSProtocol" description="TC-292" finished-at="2025-03-25T01:18:44 EET" duration-ms="19358" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[🟢 Test Started: verifyHTTPSProtocol (🔢 Test Case ID: TC-292)]]>
            </line>
            <line>
              <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
            </line>
            <line>
              <![CDATA[✅ Test PASSED: verifyHTTPSProtocol (Duration: 19358ms)]]>
            </line>
            <line>
              <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-292\TC-292_screenshot.png' height='300' width='400'/><br>]]>
            </line>
          </reporter-output>
        </test-method> <!-- verifyHTTPSProtocol -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:18:44 EET" name="tearDown" finished-at="2025-03-25T01:18:44 EET" duration-ms="362" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verifyHTTPSProtocol status=SUCCESS method=SecurityTest.verifyHTTPSProtocol()[pri:0, instance:tests.ui.SecurityTest@305f7627] output=🟢 Test Started: verifyHTTPSProtocol (🔢 Test Case ID: TC-292)]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(org.testng.ITestResult,org.testng.ITestContext)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:18:44 EET" name="setUp" finished-at="2025-03-25T01:19:01 EET" duration-ms="17291" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name={null} status=CREATED method=SecurityTest.verifyPasswordStrengthRequirements()[pri:0, instance:tests.ui.SecurityTest@305f7627] output={null}]]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[org.testng.TestRunner@6796a873]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="verifyPasswordStrengthRequirements()[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:19:01 EET" name="verifyPasswordStrengthRequirements" description="TC-034" finished-at="2025-03-25T01:19:10 EET" duration-ms="8090" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Error message did not include 'at least 8 characters' requirement. expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Error message did not include 'at least 8 characters' requirement. expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertTrue(Assert.java:57)
at tests.ui.SecurityTest.verifyPasswordStrengthRequirements(SecurityTest.java:149)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:155)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:102)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:91)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:137)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[🟢 Test Started: verifyPasswordStrengthRequirements (🔢 Test Case ID: TC-034)]]>
            </line>
            <line>
              <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
            </line>
            <line>
              <![CDATA[❌ Test FAILED: verifyPasswordStrengthRequirements - Error message did not include 'at least 8 characters' requirement. expected [true] but found [false]]]>
            </line>
            <line>
              <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-034\TC-034_screenshot.png' height='300' width='400'/><br>]]>
            </line>
          </reporter-output>
        </test-method> <!-- verifyPasswordStrengthRequirements -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:19:10 EET" name="tearDown" finished-at="2025-03-25T01:19:10 EET" duration-ms="393" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verifyPasswordStrengthRequirements status=FAILURE method=SecurityTest.verifyPasswordStrengthRequirements()[pri:0, instance:tests.ui.SecurityTest@305f7627] output=🟢 Test Started: verifyPasswordStrengthRequirements (🔢 Test Case ID: TC-034)]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(org.testng.ITestResult,org.testng.ITestContext)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:19:10 EET" name="setUp" finished-at="2025-03-25T01:19:26 EET" duration-ms="15891" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name={null} status=CREATED method=SecurityTest.verifyPaymentGatewaySecurity()[pri:0, instance:tests.ui.SecurityTest@305f7627] output={null}]]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[org.testng.TestRunner@6796a873]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="verifyPaymentGatewaySecurity()[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:19:26 EET" name="verifyPaymentGatewaySecurity" description="TC-091" finished-at="2025-03-25T01:19:58 EET" duration-ms="32492" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Card number is stored in plaintext. It should be encrypted or masked. expected [false] but found [true]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Card number is stored in plaintext. It should be encrypted or masked. expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertFalse(Assert.java:79)
at tests.ui.SecurityTest.verifyPaymentGatewaySecurity(SecurityTest.java:186)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:155)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:102)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:91)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:137)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[🟢 Test Started: verifyPaymentGatewaySecurity (🔢 Test Case ID: TC-091)]]>
            </line>
            <line>
              <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
            </line>
            <line>
              <![CDATA[❌ Test FAILED: verifyPaymentGatewaySecurity - Card number is stored in plaintext. It should be encrypted or masked. expected [false] but found [true]]]>
            </line>
            <line>
              <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-091\TC-091_screenshot.png' height='300' width='400'/><br>]]>
            </line>
          </reporter-output>
        </test-method> <!-- verifyPaymentGatewaySecurity -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:19:59 EET" name="tearDown" finished-at="2025-03-25T01:19:59 EET" duration-ms="459" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verifyPaymentGatewaySecurity status=FAILURE method=SecurityTest.verifyPaymentGatewaySecurity()[pri:0, instance:tests.ui.SecurityTest@305f7627] output=🟢 Test Started: verifyPaymentGatewaySecurity (🔢 Test Case ID: TC-091)]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(org.testng.ITestResult,org.testng.ITestContext)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:19:59 EET" name="setUp" finished-at="2025-03-25T01:20:15 EET" duration-ms="16026" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name={null} status=CREATED method=SecurityTest.verifyProtectionAgainstXSS()[pri:0, instance:tests.ui.SecurityTest@305f7627] output={null}]]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[org.testng.TestRunner@6796a873]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="verifyProtectionAgainstXSS()[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:20:15 EET" name="verifyProtectionAgainstXSS" description="TC-126" finished-at="2025-03-25T01:20:16 EET" duration-ms="1028" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[XSS attack successful: payload is reflected in the Email field. expected [false] but found [true]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: XSS attack successful: payload is reflected in the Email field. expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1590)
at org.testng.Assert.assertFalse(Assert.java:79)
at tests.ui.SecurityTest.verifyProtectionAgainstXSS(SecurityTest.java:277)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:155)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:102)
at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:91)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:137)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[🟢 Test Started: verifyProtectionAgainstXSS (🔢 Test Case ID: TC-126)]]>
            </line>
            <line>
              <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
            </line>
            <line>
              <![CDATA[❌ Test FAILED: verifyProtectionAgainstXSS - XSS attack successful: payload is reflected in the Email field. expected [false] but found [true]]]>
            </line>
            <line>
              <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-126\TC-126_screenshot.png' height='300' width='400'/><br>]]>
            </line>
          </reporter-output>
        </test-method> <!-- verifyProtectionAgainstXSS -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:20:16 EET" name="tearDown" finished-at="2025-03-25T01:20:16 EET" duration-ms="211" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verifyProtectionAgainstXSS status=FAILURE method=SecurityTest.verifyProtectionAgainstXSS()[pri:0, instance:tests.ui.SecurityTest@305f7627] output=🟢 Test Started: verifyProtectionAgainstXSS (🔢 Test Case ID: TC-126)]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(org.testng.ITestResult,org.testng.ITestContext)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:20:16 EET" name="setUp" finished-at="2025-03-25T01:20:33 EET" duration-ms="16954" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name={null} status=CREATED method=SecurityTest.verifySessionHandling()[pri:0, instance:tests.ui.SecurityTest@305f7627] output={null}]]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[org.testng.TestRunner@6796a873]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="verifySessionHandling()[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:20:33 EET" name="verifySessionHandling" description="TC-021" finished-at="2025-03-25T01:20:51 EET" duration-ms="17272" status="PASS">
          <reporter-output>
            <line>
              <![CDATA[🟢 Test Started: verifySessionHandling (🔢 Test Case ID: TC-021)]]>
            </line>
            <line>
              <![CDATA[
🌍 Test Environment:
🖥️ OS: Windows 11 10.0
☕ Java Version: 23.0.1
🌐 Browser: chrome 134.0.6998.118
]]>
            </line>
            <line>
              <![CDATA[✅ Test PASSED: verifySessionHandling (Duration: 17272ms)]]>
            </line>
            <line>
              <![CDATA[<br><img src='E:/Folder/Galal/Courses/Digital Egypt Pioneers Initiative (DEPI) - Software Testing/DEPI Software Tester Graduation Project/automation-framework/reports/screenshots\TC-021\TC-021_screenshot.png' height='300' width='400'/><br>]]>
            </line>
          </reporter-output>
        </test-method> <!-- verifySessionHandling -->
        <test-method is-config="true" signature="tearDown(org.testng.ITestResult)[pri:0, instance:tests.ui.SecurityTest@305f7627]" started-at="2025-03-25T01:20:51 EET" name="tearDown" finished-at="2025-03-25T01:20:51 EET" duration-ms="208" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verifySessionHandling status=SUCCESS method=SecurityTest.verifySessionHandling()[pri:0, instance:tests.ui.SecurityTest@305f7627] output=🟢 Test Started: verifySessionHandling (🔢 Test Case ID: TC-021)]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- tests.ui.SecurityTest -->
    </test> <!-- Surefire test -->
  </suite> <!-- Surefire suite -->
</testng-results>
